crates/astria-core/src/protocol/transaction/v1alpha1/mod.rs

```diff
@@ -381,35 +417,50 @@ enum UnsignedTransactionErrorKind {
         raw::UnsignedTransaction::type_url()
     )]
     DecodeAny(#[source] prost::DecodeError),
+    #[error("`actions` field does not make a valid `ActionGroup`")]
+    ActionGroup(#[source] action_group::Error),
 }
 
-pub struct TransactionParamsBuilder<TChainId = std::borrow::Cow<'static, str>> {
+pub struct UnsignedTransactionBuilder<TChainId = std::borrow::Cow<'static, str>> {
     nonce: u32,
     chain_id: TChainId,
+    actions: Vec<Action>,
 }
 
-impl TransactionParamsBuilder {
-    fn new() -> Self {
+impl UnsignedTransactionBuilder {
+    #[must_use]
+    #[allow(clippy::new_without_default)]
```

Comment:
We should be able to just derive `Default` on the builder rather than doing this `allow` I think.